File: backend/src/main/scala/microservices/operations/overall/objects/ManagerDecisionPlannerRequest.scala
// 本文件定义 `ConfirmManagerBookingItemPlanner` 和 `RejectManagerBookingItemPlanner` 对应的请求对象。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerDecisionPlannerRequest(managerId: String, managerType: String, orderItemId: String, reason: Option[String], note: Option[String])
object ManagerDecisionPlannerRequest:
  given sourceEncoder: Encoder[ManagerDecisionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ManagerDecisionPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/overall/objects/ManagerDecisionPlannerRequest.ts
// 本文件定义 `ConfirmManagerBookingItemPlanner` 和 `RejectManagerBookingItemPlanner` 的请求对象。

export type ManagerDecisionPlannerRequest = {
  managerId: string
  managerType: string
  orderItemId: string
  reason?: string | null
  note?: string | null
}

export const managerDecisionPlannerRequestFromJson = (json: string): ManagerDecisionPlannerRequest =>
  JSON.parse(json) as ManagerDecisionPlannerRequest

export const managerDecisionPlannerRequestToJson = (value: ManagerDecisionPlannerRequest): string =>
  JSON.stringify(value)
